svelte-case-study.md

```diff
@@ -0,0 +1,350 @@
+
+{@debug user}
+
+<h1>Hello {user.firstname}!</h1>
+
+<style>
+h1 {
+color: green;
+}
+</style>
+```
+This will debug `user` object and console log its response.
+
+#### [Learn more](https://svelte.dev/docs#template-syntax-debug)
+---
+`@const myVar` defines a local constant and is only allowed as a direct child of 
+`{#if}`, `{:else if}`, `{:else}`, `{#each}`, `{:then}`, `{:catch}`, `<Component />` or `<svelte:fragment />`.
+
+For example:
+```js
+<script>
+	export let boxes;
+</script>
+
+{#each boxes as box}
+	{@const area = box.width * box.height}
+	{box.width} * {box.height} = {area}
+{/each}
+```
+
+#### [Learn more](https://svelte.dev/docs#template-syntax-const)
+---
+`{#each items as item} {/each}` is used to loop through an array or object. You can also include index, a key value for each loop, and `{:else}`.
+
+For example:
+```js
+<h1>Shopping list</h1>
+<ul>
+	{#each items as item}
+		<li>{item.name} x {item.qty}</li>
+	{/each}
+</ul>
+```
+
+#### [Learn more](https://svelte.dev/docs#template-syntax-each)
+---
+`{#await expression}.. {:then item} .. {:catch error} .. {/await}` is used to handle promises in Svelte. You can also exclude `:then`, and if you do not need to render anything when the promise rejects, you can also remove `:catch`.
+
+For example:
+```js
+ <div class="items-container">
+   <ul>
+     {#await fetchItems(baseUrl + API_KEY + langEnUs)}
+<---- Promise is pending ----->
+       <p style="font-weight: bold">loading...</p>
+     {:then items}
+<---- Promise was fulfilled ----->
+       {#each items as item}
+         <li id={item.id}>{item.name}</li>
+       {/each}
+     {:catch err}
+<---- Promise was rejected ----->
+       <p class="error">Something went wrong! -- {err}</p>
+     {/await}
+   </ul>
+ </div>
+```
+
+#### [Learn more](https://svelte.dev/docs#template-syntax-await).
+---
+
+[Svelte template syntax](https://svelte.dev/docs#template-syntax)
+
+## Strengths
+
+- The build time is much faster when compared to react or even other frameworks.
+
+- The bundle size is smaller and tiny when gzipped.
+
+- Binding classes and variables are relatively easy, and custom logic is not needed when binding classes.
+
+- Scoping CSS withing the component itself allows flexible styling.
+
+- It is easier to understand and get started with compared to other frameworks.
+
+## Weaknesses
+
+- Svelte does not automatically listen for updates to references and array mutations, so developers need to actively watch for these and ensure that arrays are reassigned in order to upodate the UI.
+
+- The syntax for handling DOM events can also be annoying, as it requires special syntax such as `on:click` rather than more straight forward like `onClick`.
+
+- Svelte is a new and relatively young framework with minimal community support, thereby does not have support for a wide range of plugins and integrations that might be required by large-scale production applications. In this regard, React is the powerful contender here.
```

Comment:
This is the real kicker, and a bit of a chicken and egg situation too.

REPL is a great concept but I don't think it clicked with the community.